nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/SolrProcessor.java

```diff
@@ -275,7 +275,7 @@ protected final boolean isBasicAuthEnabled() {
     }
 
     @Override
-    protected final Collection<ValidationResult> customValidate(ValidationContext context) {
+    protected Collection<ValidationResult> customValidate(ValidationContext context) {
```

Comment:
Shouldn't we add another protected method to override at sub-classes?

Reply:
I did within class GetSolr

Reply:
I imagine the reason why this customValidate is marked with `final` is that because the original author wanted to avoid sub-classes skip executing validation code implemented here. You implemented within GetSolr, and call `super.customValidate` from there, so it should be fine, but other sub-class can forget to call `super.customValidate` if we remove `final` keyword.

So, I thought it might be safer approach to add an abstract method, such as `additionalCustomValidate` at SolrProcessor, then call it from customValidate, and let sub-classes implement custom validation in it.

Reply:
ok. by doing so, i will also have to add this method to PutSolrContentStream

Reply:
Good call. Then the method can be a non-abstract method at SolrProcessor that does nothing.